core/src/wsv.rs

```diff
@@ -33,53 +27,112 @@ pub type NewBlockNotificationSender = tokio::sync::watch::Sender<()>;
+        /// Registered domains.
+        domains: DomainsMap,
+        /// Roles. [`Role`] pairs.
+        roles: crate::RolesMap,
+        /// Triggers
+        triggers: TriggerSet,
+    }
+
+    impl World {
+        /// Creates an empty `World`.
+        pub fn new() -> Self {
+            Self::default()
+        }
 
-/// The global entity consisting of `domains`, `triggers` and etc.
-/// For example registration of domain, will have this as an ISI target.
-#[derive(Debug, Default, Clone)]
-pub struct World {
-    /// Iroha parameters.
-    pub parameters: Vec<Parameter>,
-    /// Identifications of discovered trusted peers.
-    pub trusted_peers_ids: PeersIds,
-    /// Registered domains.
-    pub domains: DomainsMap,
-    /// Roles. [`Role`] pairs.
-    pub roles: crate::RolesMap,
-    /// Triggers
-    pub triggers: TriggerSet,
-}
+        /// Creates a [`World`] with these [`Domain`]s and trusted [`PeerId`]s.
+        pub fn with<D, P>(domains: D, trusted_peers_ids: P) -> Self
+        where
+            D: IntoIterator<Item = Domain> + 'static,
+            P: IntoIterator<Item = PeerId> + 'static,
+        {
+            let domains = domains
+                .into_iter()
+                .map(|domain| (domain.id().clone(), domain))
+                .collect();
+            let trusted_peers_ids = trusted_peers_ids.into_iter().collect();
+            World {
+                domains,
+                trusted_peers_ids,
+                ..World::new()
+            }
+        }
+    }
 
-impl Deref for World {
-    type Target = Self;
-    #[inline]
-    fn deref(&self) -> &Self::Target {
-        self
+    #[cfg(feature = "mock_world")]
+    mockall::mock! {
+        pub World {
+            /// Creates an empty `World`.
+            pub fn new() -> Self;
+
+            /// Creates a [`World`] with these [`Domain`]s and trusted [`PeerId`]s.
+            pub fn with<D, P>(domains: D, trusted_peers_ids: P) -> Self
+            where
+                D: IntoIterator<Item = Domain> + 'static,
+                P: IntoIterator<Item = PeerId> + 'static;
+
+
+            /// Returns [`World`] parameters
+            pub fn parameters(&self) -> &Vec<Parameter>;
+
+            /// Returns [`World`] trusted peers ids
+            pub fn trusted_peers_ids(&self) -> &PeersIds;
+
+            /// Returns [`World`] domains
+            pub fn domains(&self) -> &DomainsMap;
+
+            /// Returns [`World`] roles
+            pub fn roles(&self) -> &crate::RolesMap;
+
+            /// Returns [`World`] triggers
+            pub fn triggers(&self) -> &TriggerSet;
+        }
+
+        impl std::fmt::Debug for World {
+            fn fmt<'fm>(&self, f: &mut std::fmt::Formatter<'fm>) -> std::fmt::Result;
+        }
+
+        impl Clone for World {
+            fn clone(&self) -> Self;
+        }
     }
 }
 
-impl DerefMut for World {
-    #[inline]
-    fn deref_mut(&mut self) -> &mut Self::Target {
-        self
+cfg_if::cfg_if! {
```

Comment:
checkout [mockall_double](https://docs.rs/mockall_double/0.3.0/mockall_double/)

Reply:
Yeah, I know about it, but I also need to `pub use World as OriginalWorld` which `mockall_double` does not provide

Reply:
okay